topdown/graphql.go

```diff
@@ -453,10 +453,38 @@ func builtinGraphQLIsValid(_ BuiltinContext, operands []*ast.Term, iter func(*as
 	return iter(ast.BooleanTerm(true))
 }
 
+func builtinGraphQLIsSchemaValid(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+	var schemaDoc *gqlast.SchemaDocument
+	var err error
+
+	switch x := operands[0].Value.(type) {
+	case ast.String:
+		schemaDoc, err = parseSchema(string(x))
+	case ast.Object:
+		schemaDoc, err = objectToSchemaDocument(x)
+	default:
+		// Error if wrong type.
+		return iter(ast.BooleanTerm(false))
+	}
+	if err != nil {
+		return iter(ast.BooleanTerm(false))
+	}
+
+	// Validate the schema, erroring if there's an issue.
+	_, err = convertSchema(schemaDoc)
+	if err != nil {
+		return iter(ast.BooleanTerm(false))
+	}
+
+	// If we got this far, the GraphQL query passed validation.
+	return iter(ast.BooleanTerm(true))
```

Comment:
[golf] (let's not do that 😄)
```suggestion
	return iter(ast.BooleanTerm(err != nil))
```

Reply:
Rolled in the suggestion here. Marking as resolved. :+1: